File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    x_term = 0
    constant_term = 0
    terms = polynomial.split(" + ")
    
    for term in terms:
        if 'x' in term:
            if term == 'x':
                x_term += 1
            else:
                x_term += int(term.replace('x', ''))
        else:
            constant_term += int(term)
    
    result = []
    
    if x_term != 0:
        if x_term == 1:
            result.append("x")
        else:
            result.append(f"{x_term}x")
    
    if constant_term != 0:
        result.append(f"{constant_term}")
        
    return " + ".join(result)
            

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.cpp
#include <string>
#include <vector>
#include <sstream>

using namespace std;

string solution(string polynomial) {
    istringstream ss(polynomial);
    string term;
    int x_term = 0;
    int constant_term = 0;
    
    while (ss >> term) {
        if (term.find('x') != string::npos) {
            if (term == "x") x_term += 1;
            else x_term += stoi(term.substr(0, term.find('x')));
        }
        else if (term != "+") constant_term += stoi(term);
    }
    
    string result;
    
    if (x_term > 0) {
        if (x_term == 1) result += "x";
        else result += to_string(x_term) + "x";
    }
    
    if (constant_term > 0) {
        if (!result.empty()) result += " + ";
        result += to_string(constant_term);
    }
    
    if (result.empty()) {
        return 0;
    }
    return result;
}